yt_dlp/__init__.py

```diff
@@ -978,6 +985,22 @@ def _real_main(argv=None):
             traceback.print_exc()
             ydl._download_retcode = 100
 
+        if opts.list_impersonate_targets:
+            available_targets = ydl.get_impersonate_targets()
+            rows = [[*[item or '' for item in target], compile_impersonate_target(*target)] for target in
+                    available_targets]
+
+            ydl.to_screen('[info] Available impersonate targets')
+            ydl.to_stdout(
+                render_table(['Client', 'Version', 'OS', 'OS Version', 'Handler', 'Example'], rows)
+            )
+            if not available_targets:
+                ydl.to_stdout('You are missing dependencies for impersonation. See the README for more info.')
+            ydl.to_stdout(
+                'If the above table is missing targets, you may be missing dependencies for impersonation. '
+                'See the documentation for more information.')
```

Comment:
see above review comment

Reply:
That would get messy I feel. You'd have to keep a separate list of supported targets somewhere, which would be a pain to maintain.

Reply:
Thinking about it, the majority of users will only care about the browser. The version etc. is probably irrelevant in most cases. Same for extractors, in most cases you would only specify the browser.

This would suggest most users would be looking for what browsers a dependency will provide.

Given this, all we might need to provide somewhere is some information on what browsers each dep supports. Which could be done in the list target table.

E.g. curl_cffi provides chrome, edge and safari